core/storage/user/gae_models.py

```diff
@@ -2068,6 +2068,83 @@ def increment_score_for_user(cls, user_id, score_category, increment_by):
             model.put()
 
 
+class UserCommunityRightsModel(base_models.BaseModel):
+    """Model for storing user's rights on community dashboard.
+
+    Instances of this class are keyed by the user id.
+    """
+    can_review_translation_for_language_codes = ndb.StringProperty(
+        repeated=True, indexed=True)
+    can_review_voiceover_for_language_codes = ndb.StringProperty(
+        repeated=True, indexed=True)
+    can_review_questions = ndb.BooleanProperty(indexed=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """The model can be deleted since it only contains information relevant
+        to the one user.
+        """
+        return base_models.DELETION_POLICY.DELETE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether UserCommunityRightsModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.get_by_id(user_id) is not None
+
+    @classmethod
+    def get_translation_reviewer_user_id(cls, language_code):
+        """Returns the ID of the users who has rights to review translation in
+        the given language code.
+
+        Args:
+            language_code: str. The code of the language.
+
+        Returns:
+            list(str). A list of user's IDs who has rights to review translation
+            in the given language code.
+        """
+        reviewer_models = (
+            cls.query(
+                cls.can_review_translation_for_language_codes == language_code)
+            .fetch())
```

Comment:
Can you do a keys_only query if all you need is the IDs? Might save datastore bandwidth.

Reply:
I think you can do that easily enough -- key.id or key.id() or similar, IIRC. Please do check.

Note that this method should return user IDs, not usernames.